includes/REST_API/Font_Controller.php

```diff
@@ -0,0 +1,586 @@
+		$post_type = get_post_type_object( $this->post_type );
+
+		if ( ! $post_type || ! current_user_can( $post_type->cap->read_post ) ) {
+			return new WP_Error(
+				'rest_forbidden',
+				__( 'Sorry, you are not allowed to list fonts.', 'web-stories' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Force-deletes a single font.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$request->set_param( 'force', true );
+
+		return parent::delete_item( $request );
+	}
+
+	/**
+	 * Prepares a single post output for response.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @param WP_Post         $item    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ): WP_REST_Response {
+		// Restores the more descriptive, specific name for use within this method.
+		$post            = $item;
+		$GLOBALS['post'] = $post; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		setup_postdata( $post );
+
+		$fields = $this->get_fields_for_response( $request );
+
+		$data = [];
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'family', $fields ) ) {
+			$data['family'] = $post->post_title;
+		}
+
+		if ( rest_is_field_included( 'service', $fields ) ) {
+			$data['service'] = 'custom';
+		}
+
+		/**
+		 * Font data.
+		 *
+		 * @var array|null $font_data
+		 */
+		$font_data = json_decode( $post->post_content, true );
+
+		if ( $font_data ) {
+			foreach ( $font_data as $key => $value ) {
+				if ( rest_is_field_included( $key, $fields ) ) {
+					$data[ $key ] = $value;
+				}
+			}
+		}
+
+		/**
+		 * Request context.
+		 *
+		 * @var string $context
+		 */
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		/**
+		 * Response object.
+		 *
+		 * @var WP_REST_Response $response
+		 */
+		$response = rest_ensure_response( $data );
+
+		$links = $this->prepare_links( $post );
+		$response->add_links( $links );
+
+		if ( ! empty( $links['self']['href'] ) ) {
+			$actions = $this->get_available_actions( $post, $request );
+
+			$self = $links['self']['href'];
+
+			foreach ( $actions as $rel ) {
+				$response->add_link( $rel, $self );
+			}
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single post for create.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
```

Comment:
Should we check that font family exists before saving. I am worried that someone will try and create two fonts with the same name. Either by uploading two fonts with the same name or by adding one that already appears in list of fonts.

Reply:
Sounds reasonable 👍